design/data-mover.md

```diff
@@ -0,0 +1,73 @@
+# Velero Data Mover
+
+## Abstract
+Velero currently does not handle data movement of snapshotted data.  We would like to have the ability to move data that has been snapshotted in a storage system or other [Astrolabe Protected Entity](https://github.com/vmware-tanzu/astrolabe) provider into the backup storage location specified.
+
+## Background
+Velero has two options for backing up persistent volumes:
+Using snapshotting APIs provided by the storage provider (for example, using the VolumeSnapshotter)
+Filesystem level backup using restic
+
+Snapshotting capabilities provided by storage providers generally provide crash consistency of volumes, however they are not portable and are typically locked to the particular cloud provider and region in which the snapshot was taken.
+Although restic provides portability of data, it is a filesystem level backup and is not crash-consistent by default.
+It requires the quiescing of data prior to backup which impacts the availability of an application.
+
+Velero also has beta support for taking snapshots using CSI.
+Without data movement though, there is no guarantee that the snapshots are durable as it is dependent on the storage provider.
+
+Extraction of data from snapshots is something that is already available in the [Velero Plugin for vSphere](https://github.com/vmware-tanzu/velero-plugin-for-vsphere).
+This plugin makes use of the Astrolabe APIs to implement a vSphere FCD Protected Entity.
+This Protected Entity interface provides data readers and writers for the disks.
+Once a snapshot is created, the data reader is used to read the data from the disk and it is written to an S3 compatible object store.
+
+The Astrolabe API can be implemented for resources other than volumes, for example, databases.
+This enables each resource to define the best way to back up that entity and any subresources that it depends on.
+For example, when backing up a Postgres database, it may be preferable to use pg_dump rather than quiescing and snapshotting the volumes.
+This can be achieved using Astrolabe and as a result will provide data readers for the dump file.
+This data reader can be used to write the dump contents to an arbitrary backup location, rather than being confined to the underlying storage system.
+
+By allowing Velero to interact with Astrolabe Protected Entities, we can utilize the data readers for these entities to read the data and write to other locations.
+
+## Goals
+- Enable the data from snapshots of resources to be moved to multiple storage locations including object stores.
+- Allow the location for the data store to be specified at backup time and allow selection of backup location at restore time.
```

Comment:
To clarify: does this include replication at backup time? For example https://github.com/vmware-tanzu/velero-plugin-for-aws/pull/90 allows you to create an EBS snapshot in one region and then replicate it to another, which means two snapshot ids must be saved in backup metadata; the appropriate id is selected at restore time according to the region in which the restore is running. In that example the backup storage location for metadata just points to one region (since you can access the blob globally and performance is not an issue).